### DISCUSSION TITLE
Documentation for `is...` methods?

### DISCUSSION THREAD

**Discussion by overlookmotel** [babel/babel#11714] [Q&A]
[@babel/types](https://babeljs.io/docs/en/babel-types) includes various `is...` methods which do not refer to a documented type.

For some it's quite obvious what they do e.g. `isLet`. For others, it's more obscure e.g. `isReferenced`, `isNodesEquivalent`.

Is there any documentation for these methods? I am hoping I'm just looking in the wrong place.

The `isX` methods with no corresponding `X` type are:

`isBinary`
`isBinding`
`isBlock`
`isBlockParent`
`isBlockScoped`
`isClass`
`isCompletionStatement`
`isConditional`
`isDeclaration`
`isEnumBody`
`isEnumMember`
`isExportDeclaration`
`isExpression`
`isExpressionWrapper`
`isFlow`
`isFlowBaseAnnotation`
`isFlowDeclaration`
`isFlowPredicate`
`isFlowType`
`isFor`
`isForXStatement`
`isFunction`
`isFunctionParent`
`isImmutable`
`isJSX`
`isLVal`
`isLet`
`isLiteral`
`isLoop`
`isMethod`
`isModuleDeclaration`
`isModuleSpecifier`
`isNode`
`isNodesEquivalent`
`isObjectMember`
`isPattern`
`isPatternLike`
`isPlaceholderType`
`isPrivate`
`isProperty`
`isPureish`
`isReferenced`
`isScopable`
`isScope`
`isSpecifierDefault`
`isStatement`
`isTSBaseType`
`isTSEntityName`
`isTSType`
`isTSTypeElement`
`isTerminatorless`
`isType`
`isUnaryLike`
`isUserWhitespacable`
`isValidES3Identifier`
`isValidIdentifier`
`isVar`
`isWhile`
`react.isCompatTag`
`react.isReactComponent`

**Comment by overlookmotel** [comment]
Also, there are some types not mentioned in [@babel/types API docs](https://babeljs.io/docs/en/babel-types), such as the `ReferencedIdentifier` type mentioned [here](https://github.com/babel/babel/discussions/11654#discussioncomment-23236). Are these documented somewhere?

**Comment by mischnic** [comment]
(All I know is that these are "virtual types" and I'm not aware of any documentation).
